sdv/constraints/tabular.py

```diff
@@ -424,10 +425,17 @@ def _validate_inputs(column_name, value, relation):
         if not (isinstance(value, (int, float)) or is_datetime_type(value)):
             raise ValueError('`value` must be a number or datetime.')
 
+        if is_datetime_type(value):
+            if not isinstance(value, str):
+                raise ValueError('Datetime must be represented as a string.')
+
+            return cast_to_datetime64(value)
+
+        return value
```

Comment:
I don't think the `_validate_inputs` method should return a value. If we want to group the logic we can make a `parse_value` method that calls `_validate_inputs` and then parses to datetime if necessary